tests/mypy/modules/plugin_success.py

```diff
@@ -104,3 +105,17 @@ class ClassVarModel(BaseModel):
 
 
 ClassVarModel(x=1)
+
+
+class Config:
+    validate_assignment = True
+
+
+@dataclass(config=Config)
+class AddProject:
```

Comment:
@dmontagu 
>  It looks to me like the received config isn't currently used by mypy, it's just allowed to be passed without raising an error -- is that right?

I understand it is right because mypy passes the tests without any error.
Mypy can parse a type declaration of `config=`. 
Of course,  mypy don't use the contents of `Config` like `orm_mode` yet.

Reply:
Yeah, I think that's fine for now. We can revisit eventually if we think it matters.